apps/site/assets/js/test/support-form_test.js

```diff
@@ -487,4 +478,56 @@ describe("support form", () => {
       assert.equal(photos.length, toUpload.length);
     });
   });
+
+  describe("handleModeChangeSelection", () => {
+    beforeEach(() => {
+      const opts = { opt1_options: [1, 2, 3], opt2_options: [4, 5] };
+
+      $("#test").html(`
+      <script id="js-routes-by-mode" type="text/plain">
+        ${JSON.stringify(opts)}
+      </script>
+       <select class="c-select c-mode-selector" id="support_mode">
+         <option value=" ">Select</option>
+         <option value="opt1">opt1</option>
+         <option value="opt2">opt2</option>
+       </select>
+       <div id="routeAndVehicle">
+        <select class="c-select c-route-selector" id="support_route"></select>
+       </div>
+     `);
+      handleModeChangeSelection($);
+    });
+
+    afterEach(() => {
+      $("#test").remove();
+    });
+
+    it("detects a new selection and updates the options with dynamic values", () => {
+      // initEvent is deprecated and no longer recommended but the newer way of triggering events nor jQuery's .change() were triggering a change in the selection
```

Comment:
hmm I wonder if this is a JSDom thing 🤔 
anyhow good on you for getting around it 😀